src/exit_handler/src/exit_handler_intel_x64_eapis_cpuid_emulation.cpp

```diff
@@ -0,0 +1,55 @@
+//
+// Bareflank Extended APIs
+//
+// Copyright (C) 2015 Assured Information Security, Inc.
+// Author: Rian Quinn        <quinnr@ainfosec.com>
+// Author: Brendan Kerrigan  <kerriganb@ainfosec.com>
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+#include <exit_handler/exit_handler_intel_x64_eapis.h>
+
+void
+exit_handler_intel_x64_eapis::log_cpuid_access(bool enable)
+{ m_cpuid_access_log_enabled = enable; }
+
+void
+exit_handler_intel_x64_eapis::clear_cpuid_access_log()
+{ m_cpuid_access_log.clear(); }
+
+void
+exit_handler_intel_x64_eapis::handle_exit__cpuid()
+{
+    cpuid_key_type leaf = m_state_save->rax;
+    cpuid_key_type subleaf = m_state_save->rcx;
+    cpuid_key_type key = (leaf << 32) | subleaf;
```

Comment:
I didn't use the helper `create_key` function in this instance. The registers in m_state_save are of type uint64_t (cpuid_key_type), but the helper function uses uint32_t (cpuid_type) since that is what is normally passed as leaf / subleaf values. I thought that re-casting the m_state_save registers would be messier than just shifting the key manually in this single instance.

Reply:
you should be able to up cast without an actual cast operator